core/src/test/java/org/apache/iceberg/TestClientPoolImpl.java

```diff
@@ -69,6 +113,19 @@ public void testNoRetryingNonRetryableException() {
     }
   }
 
+  @Test
+  public void testNoRetryingNonRetryableImplementationSpecificException() {
```

Comment:
maybe name this `customNonRetryableExceptionIsNotRetried` as the name is quite difficult to decipher